x/authz/codec/cdc.go

```diff
@@ -0,0 +1,18 @@
+package codec
```

Comment:
I've created this new package in order to avoid any possible dependency cycle that might arise if we leave the `ModuleCdc` definition inside the `x/authz/codec.go` file. If you think we can keep it inside `x/authz/codec.go` then I can move it there without any issue

Reply:
>  to avoid any possible dependency cycle that might arise if we leave the ModuleCdc definition inside the x/authz/codec.go

Which dep cycle do you see happening?

By default, I would say let's keep like all other modules, unless there's a strong reason to do a separate package.

Reply:
Ok I've just tried moving it back to `authz/codec.go` and the following popped up: 
```
package github.com/cosmos/cosmos-sdk/server
        imports github.com/cosmos/cosmos-sdk/server/rosetta
        imports github.com/cosmos/cosmos-sdk/x/auth/tx
        imports github.com/cosmos/cosmos-sdk/x/auth/middleware
        imports github.com/cosmos/cosmos-sdk/x/auth/types
        imports github.com/cosmos/cosmos-sdk/x/authz
        imports github.com/cosmos/cosmos-sdk/x/auth/types: import cycle not allowed
package github.com/cosmos/cosmos-sdk/server
        imports github.com/cosmos/cosmos-sdk/server/rosetta
        imports github.com/cosmos/cosmos-sdk/x/auth/tx
        imports github.com/cosmos/cosmos-sdk/x/auth/middleware
        imports github.com/cosmos/cosmos-sdk/x/auth/types
        imports github.com/cosmos/cosmos-sdk/x/authz
        imports github.com/cosmos/cosmos-sdk/x/auth/types: import cycle not allowed
```

Reply:
Nice! Can we add a comment somewhere saying that authz's codec is in a separate package to avoid dep cycle? Maybe a package godoc

Reply:
@amaurym @alexanderbez documentation has been added